Add support for new cluster bootstrapping

jsravn opened the pull request.

jsravn commented:
Vendored stuff is a separate commit, so the changes can be reviewed easily.

chbatey commented on main.go:
```diff
@@ -1,4 +1,31 @@
+	out += bootstrapper.Bootstrap()
+
+	log.Infof("Writing environment variables to %s", outputFilename)
+	ioutil.WriteFile(outputFilename, []byte(out), 0644)
```
check error and return a non 0 if it has failed so etcd doesn't start up

chbatey commented on lib/etcdcluster/etcdcluster.go:
```diff
@@ -0,0 +1,52 @@
+
+	c, err := client.New(client.Config{Endpoints: endpoints})
+	if err != nil {
+		log.Fatal("Unable to create etcd client: ", err)
```
Does Fatal panic? Otherwise can we return an error so the caller knows this hasn't worked

chbatey replied:
Ah it does. This makes sense then. I should use this more :)

chbatey commented on lib/etcdcluster/etcdcluster.go:
```diff
@@ -0,0 +1,52 @@
+
+func (e *cluster) Members() []string {
+	membersAPI := client.NewMembersAPI(e.client)
+	members, err := membersAPI.List(context.Background())
```
Might be better to use the TODO context rather than Background

jsravn replied:
as discussed, TODO is for stub contexts intended to be later replaced. In this case it is the top level context we want (Background).

chbatey commented on README.md:
```diff
@@ -9,16 +9,36 @@ It's intended for use with AWS Auto Scaling groups and etcd2.
 1. Query the ASG for all the instance IPs.
 2. Determine if joining an existing cluster or not, by querying all instances
    to see if etcd is available.
 3. Setup etcd arguments to correctly bootstrap with all the other instances.
```
Worth adding it does this by writing env vars to a file?

jsravn commented:
@chbatey I've addressed the review comments, and also I modified `bootstrapExisting` to not panic so we can start using etcd-bootstrap right away (just means that recreating a node/expanding the cluster isn't supported yet).
